src/Nethermind/Nethermind.JsonRpc/Modules/Subscribe/SubscriptionFactory.cs

```diff
@@ -54,11 +58,11 @@ public class SubscriptionFactory : ISubscriptionFactory
             _ethSyncingInfo = ethSyncingInfo ?? throw new ArgumentNullException(nameof(ethSyncingInfo));
             _specProvider = specProvider ?? throw new ArgumentNullException(nameof(specProvider));
         }
-        public Subscription CreateSubscription(IJsonRpcDuplexClient jsonRpcDuplexClient, SubscriptionType subscriptionType, Filter? filter)
+        public Subscription CreateSubscription(IJsonRpcDuplexClient jsonRpcDuplexClient, string subscriptionType, Filter? filter)
         {
             switch (subscriptionType)
             {
-                case SubscriptionType.NewHeads: 
+                case SubscriptionType.NewHeads:
```

Comment:
Question mark - maybe just get rid of this switch altogether and just register standard subscriptions in _customSubscriptions (rename to _subscriptionFactories?) in constructor